src/common/cvt.cpp

```diff
@@ -1033,6 +1058,964 @@ void CVT_string_to_datetime(const dsc* desc,
+			{
+				int month = roman_to_int(str, strLength, strOffset);
+				if (month == 0 || month > 12)
+				{
+					cb->err(Arg::Gds(isc_value_for_pattern_is_out_of_range) <<
+						string(pattern.data(), pattern.length()) << Arg::Num(1) << Arg::Num(12));
+				}
+
+				outTimes.tm_mon = month - 1;
+				return;
+			}
+			break;
+
+		case 'D':
+			if (pattern == "DD")
+			{
+				int day = parse_string_to_get_int(str, strLength, strOffset, 2);
+				if (day == 0 || day > 31)
+				{
+					cb->err(Arg::Gds(isc_value_for_pattern_is_out_of_range) <<
+						string(pattern.data(), pattern.length()) << Arg::Num(1) << Arg::Num(31));
+				}
+
+				outTimes.tm_mday = day;
+				return;
+			}
+			break;
+
+		case 'J':
+			if (pattern == "J")
+			{
+				int JDN = parse_string_to_get_int(str, strLength, strOffset, strLength - strOffset);
+
+				constexpr int minJDN = 1721426; // 0.0.0
+				constexpr int maxJDN = 5373484; // 31.12.9999
+				if (JDN < minJDN || JDN > maxJDN)
+				{
+					cb->err(Arg::Gds(isc_value_for_pattern_is_out_of_range) <<
+						string(pattern.data(), pattern.length()) << Arg::Num(minJDN) << Arg::Num(maxJDN));
+				}
+
+				int year, month, day;
+				NoThrowTimeStamp::convertJulianDateToGregorianDate(JDN, year, month, day);
+				outTimes.tm_year = year - 1900;
+				outTimes.tm_mon = month - 1;
+				outTimes.tm_mday = day;
+				return;
+			}
+			break;
+
+		case 'H':
+			if (pattern == "HH"||
+				pattern == "HH12")
+			{
+				int hours = parse_string_to_get_int(str, strLength, strOffset, 2);
+				if (hours > 12)
+				{
+					cb->err(Arg::Gds(isc_value_for_pattern_is_out_of_range) <<
+						string(pattern.data(), pattern.length()) << Arg::Num(0) << Arg::Num(12));
+				}
+
+				if (str[strOffset] == ' ')
+					strOffset++;
+
+				std::string_view period = parse_string_to_get_first_word(str, strLength, strOffset, 2);
+				if (period == "AM")
+				{
+					outTimes.tm_hour = hours == 12 ? 0 : hours;
+					return;
+				}
+				else if (period == "PM")
+				{
+					outTimes.tm_hour = hours == 12 ? hours : 12 + hours;
+					return;
+				}
+
+				cb->err(Arg::Gds(isc_incorrect_hours_period) << string(period.data(), period.length()));
+			}
+			else if (pattern == "HH24")
+			{
+				int hours = parse_string_to_get_int(str, strLength, strOffset, 2);
+				if (hours > 23)
+				{
+					cb->err(Arg::Gds(isc_value_for_pattern_is_out_of_range) <<
+						string(pattern.data(), pattern.length()) << Arg::Num(0) << Arg::Num(23));
+				}
+
+				outTimes.tm_hour = hours;
+				return;
+			}
+			break;
+
+		case 'S':
+			if (pattern == "SS")
+			{
+				int seconds = parse_string_to_get_int(str, strLength, strOffset, 2);
+				if (seconds > 59)
+				{
+					cb->err(Arg::Gds(isc_value_for_pattern_is_out_of_range) <<
+						string(pattern.data(), pattern.length()) << Arg::Num(0) << Arg::Num(59));
+				}
+
+				outTimes.tm_sec = seconds;
+				return;
+			}
+			else if (pattern == "SSSSS")
+			{
+				constexpr int maximumSecondsInDay = 23 * 60 * 60 + 59 * 60 + 59;
```

Comment:
I don't understand that name/value. Max seconds in a day is `24 * 60 * 60`.
You write it with -1, maybe ok, but isn't clear to use `24 * 60 * 60 - 1` instead?